src/content/components/Uplift/Uplift.js

```diff
@@ -20,6 +20,69 @@ export class Uplift extends React.PureComponent {
     };
   }
 
+  async fetchBugs() {
+    const trackingField = `cf_tracking_firefox${this.props.prevRelease}`;
+    const statusField = `cf_status_firefox${this.props.prevRelease}`;
+    const statusNightly = `cf_status_firefox${this.props.release}`;
+    const prevRelease = this.props.prevRelease;
+    function getFlagQuery(type) {
+      return {
+        include_fields: columns.concat([
+          trackingField,
+          statusField,
+          statusNightly,
+        ]),
+        component: BUGZILLA_TRIAGE_COMPONENTS,
+        target_milestone: ["---", `firefox ${prevRelease + 1}`],
+        order: "changeddate DESC",
+        custom: {
+          "flagtypes.name": { substring: `approval-mozilla-beta${type}` },
+        },
+      };
+    }
+    const betakey = `cf_tracking_firefox${prevRelease}`;
+    const result = await this.context.qm.runQueries([
+      getFlagQuery("?"),
+      getFlagQuery("-"),
+      getFlagQuery("+"),
+      {
+        include_fields: columns.concat([
+          trackingField,
+          statusField,
+          statusNightly,
+        ]),
+        component: BUGZILLA_TRIAGE_COMPONENTS,
+        order: "changeddate DESC",
+        custom: {
+          [betakey]: { anyexact: ["?", "+", "blocking"] },
+          "flagtypes.name": { notsubstring: `approval-mozilla-beta` },
+        },
+      },
+    ]);
+    this.setState({
```

Comment:
same kind of situation here, this state transformation could be moved into a method. then both fetchBugs and componentWillMount can share it, so we won't need to duplicate all this text, and I'll be able to read the whole PR without having a brain aneurysm 😂